src/blockdata/script.rs

```diff
@@ -271,6 +272,11 @@ impl Script {
         Script::new_witness_program(::bech32::u5::try_from_u8(0).unwrap(), &script_hash.to_vec())
     }
 
+    /// Generates P2TR-type of scriptPubkey
+    pub fn new_v1_tr(public_key: &schnorr::PublicKey) -> Script {
+        Script::new_witness_program(::bech32::u5::try_from_u8(1).unwrap(), &public_key.serialize().to_vec())
```

Comment:
Not sure why you are calling allocating `to_vec()` function just to take the reference after.

Also will need to get rid out of `u5`-caused `unwrap`, but that requires #563 merge as you said, with its `WitnessVersion` type

Reply:
You're right. I've removed that and added a commit to do the same for new_v0_wpkh() and new_v0_wsh()